lib/perl/Genome/InstrumentData/AlignmentResult.pm

```diff
@@ -1661,8 +1660,14 @@ sub revivified_alignment_bam_file_paths {
         return @bams;
     }
 
-    my $revivified_bam = File::Spec->join($p{disk_allocation}->absolute_path, 'all_sequences.bam');
-    my $merged_bam    = $self->get_merged_bam_to_revivify_per_lane_bam;
+    my $temp_allocation = $self->_get_temp_allocation($self->output_dir);
+    UR::Context->process->add_observer(
```

Comment:
Yeah, your logic is why we elected to maintain allocation ownership in the caller to begin with, so I totally agree.

I'm unsure as to what you mean by 'commit the deletion of the allocation as well', I believe that is what we are doing on line `1666`.

Thanks for the links, I do believe we may want `pre-commit` -- I will switch it unless you have a better suggestion.